### DISCUSSION TITLE
Brick + ghcid

### DISCUSSION THREAD

**Discussion by NikolaVeber** [jtdaugherty/brick#494] [Q&A]
Hi,

I am a Haskell beginner, trying to create my first interactive Haskell app with BRICK. 
My idea is to try and use ghcid for instant recompile/reload. (Ideally, it would save the app state and restart with the same state).

Is there a "standard" way to do this? 

I have tried wunning with:
> ghcid "--command=cabal run" --test=:main --reload ./app/Main.hs
Getting an issue similar to [#164](https://github.com/crytic/echidna/issues/164) "getTerminalAttributes: illegal operation (Inappropriate ioctl for device)"

Any help would be appreciated!

**Comment by jtdaugherty** [comment]
Hi @NikolaVeber - if I recall, the purpose of `ghcid` is to rebuild your program continuously, not to run it. It appears that you're trying to have `ghcid` run the program with `--command=...`. Have you tried just `ghcid` in your project directory?

**Reply by NikolaVeber** [reply]
Hi @jtdaugherty ,

thanks for the quick reply, and sorry for getting back to it so late..
I guess you are correct - I came up with a simple hack, allowing for automatic recompile/restart once a program exits. It does most of what I wanted : some sort of quick reload:
```
#!/bin/bash

while true;
do
 if ! cabal run;
 then
    echo "failed" 
    inotifywait -r . --quiet 
 fi
 echo "cabal run... restarting..." >&2
 echo "Press Ctrl-C to quit." && sleep 1
done
```
